### ISSUE TITLE
Question：how can I receive the notification netconf-session-start when I subscribe？

### ISSUE THREAD

**Issue by Julian-younger** [CESNET/netopeer2#777]
hi,michal
   How can I receive the notification 'netconf-session-start' of current session when I subscribe？and not a info "[INF]: SR: There are no subscribers for "ietf-netconf-notifications" notifications."

**Comment by michalvasko** [comment]
I am not sure I understand. You cannot receive `netconf-session-start` notification for the session that was started because you can subscribe to notifications only after your session has started. But you can subscribe with a started session and if you then start another session, the other session will receive the notification.

**Comment by Julian-younger** [comment]
yes，that is my point.Another question,When I finish subscribing the notification，I want to send a special notification to client session,But  sometimes it fails, I can get an error "nc ERROR: Session 2: received a <notification> but session is not subscribed.";
it's seams that notification reach first then the rpc call. Have any idea ?Thanks a lot.

**Comment by michalvasko** [comment]
Can you please be more specific about what it is you are doing? Specific commands/API calls and so on.

**Comment by Julian-younger** [comment]
I use API  sr_rpc_subscribe_tree to register my callback.
In my callback funcition,I  try to use  sr_event_notif_send  and nc_server_notif_send to send notification, but both API cannot work well.

**Comment by michalvasko** [comment]
Okay, please do not call `nc_server_notif_send()` from your code, that is causing the error. But `nc_event_notif_send()` should work, your NETCONF client will receive it only if it subscribed to it. Also, you should not need to link *libnetconf2* to your app and I am wondering how did you manage to send the NETCONF notification directly to the client but I would say you are doing a lot of redundant work.

**Comment by Julian-younger** [comment]
My origin requirement is when I subscribe notification in clients,the server can send a special notification to the client.
My first try was using sr_rpc_subscribe_tree() to register a callback using the xpath "/notifications:create-subscription",which has a different priority .And used sr_event_notif_send() to send the notification in the callback. But it worked not very well.Sometimes,I can get the error "nc ERROR: Session 2: received a but session is not subscribed."  
So did other tryies.

**Comment by michalvasko** [comment]
Right, because you must have subscribed with higher priority meaning at the moment you are sending the notification, *netopeer2-server* callback was not yet called and the client is not yet subscribed. So it will not work this way and I am not sure how you can achieve this without modifying the server code. I can think only of postponing sending the notification but that would be some additional work because your callback itself must return, then wait some time, and then send the notification so that the client is subscribed by that time. But it is not exactly a nice or robust solution.

**Comment by Julian-younger** [comment]
Yes，I put the code sr_rpc_subscribe_tree() in server side，I agree with your opinion,maybe I can use a timer or thread to delay some time, but it is a problem that how much time I  must postpone.

**Comment by michalvasko** [comment]
Okay, I have adjusted *netopeer2-cli* a but and now it should be able to receive notifications right when it tries to subscribe. That means you should be able to use `nc_server_notif_send()` and it should work. But I am still wondering how you managed to get a NETCONF session in your application outside of *netopeer2-server*.

**Comment by Julian-younger** [comment]
Thanks for your quick response. I merge the changes into netopeer2-cli code,and I use nc_server_notif_send() API  and  it seams work well.

> "But I am still wondering how you managed to get a NETCONF session in your application outside of netopeer2-server."
I am not sure I understand. The application is a 3rd nms, so far,I do not know how it will work.

**Comment by michalvasko** [comment]
All I am wondering is how you got the `session` that `nc_server_notif_send()` requires (assuming you are using *netopeer2* as the server) but it does not really matter, as long as it is working for you now.

**Comment by Julian-younger** [comment]
yes，I am using netopeer2 as the server,the session is just the nc_session.
